examples/next-openai/app/function-calling/page.tsx

```diff
@@ -0,0 +1,205 @@
+          }
+        ]
+      }
+      return Promise.resolve(functionResponse)
+    } else if (functionCall.name === 'get_current_time') {
+      // Get current time
+      const time = new Date().toLocaleTimeString()
+      const functionResponse: ChatRequest = {
+        messages: [
+          ...chatMessages,
+          {
+            id: nanoid(),
+            name: 'get_current_time',
+            role: 'function' as const,
+            content: JSON.stringify({ time })
+          }
+        ],
+        // You can return a list of functions here that the model can call next
+        functions
+      }
+
+      return Promise.resolve(functionResponse)
+    } else if (functionCall.name === 'eval_code_in_browser') {
+      if (functionCall.arguments) {
+        // Parsing here does not always work since it seems that some characters in generated code aren't escaped properly.
+        const parsedFunctionCallArguments: { code: string } = JSON.parse(
+          functionCall.arguments
+        )
+        const functionResponse = {
+          messages: [
+            ...chatMessages,
+            {
+              id: nanoid(),
+              name: 'eval_code_in_browser',
+              role: 'function' as const,
+              content: JSON.stringify(eval(parsedFunctionCallArguments.code))
+            }
+          ]
+        }
+        return Promise.resolve(functionResponse)
+      }
+    }
+  }
+
+  const { messages, input, handleInputChange, handleSubmit } = useChat({
+    api: '/api/chat-with-functions',
+    onFunctionCall: functionCallHandler
+  })
+
+  // Generate a map of message role to text color
+  const roleToColorMap: Record<Message['role'], string> = {
+    system: 'red',
+    user: 'black',
+    function: 'blue',
+    assistant: 'green'
+  }
+
+  const getRenderedMessage = (m: Message) => {
+    if (m.content === '') {
+      if (typeof m.function_call === 'string') {
+        return (
+          (
+            <>
+              <p>**** Function Call! **** </p>
+              {m.function_call.split('\\n').map((line, index) => (
+                <p key={index}>{line}</p>
+              ))}
+            </>
+          ) ?? ''
+        )
+      } else {
+        return (
+          (
+            <>
+              <p>**** {m.function_call?.name!} Function Call! **** </p>
+              {m.function_call?.arguments!.split('\\n').map((line, index) => (
+                <p key={index}>{line}</p>
+              ))}
+            </>
+          ) ?? ''
+        )
+      }
+    } else {
+      return m.content
+    }
+  }
+
+  return (
+    <div className="flex flex-col w-full max-w-md py-24 mx-auto stretch">
+      {messages.length > 0
+        ? messages.map((m: Message) => (
+            <div
+              key={m.id}
+              className="whitespace-pre-wrap"
+              style={{ color: roleToColorMap[m.role] }}
+            >
+              <strong>{`${
+                m.role.charAt(0).toUpperCase() + m.role.slice(1)
```

Comment:
Can we just simplify this? It doesn't add to example and is intimidating for people to read

Reply:
Sure, I'll clean up the example a bit.